### PR TITLE
Fix linked client contact user can not see existing samples

### PR DESCRIPTION
**Pull request by ramonski**
## Description of the issue/feature this PR addresses

This PR fixes an issue where linked client contact users did not see client contents (samples) prior it was linked.
Furthermore, it fixes the styling of the contact login-details view.

## Current behavior before PR

Previous created samples not visible to new linked client contact users

## Desired behavior after PR is merged

Previous created samples visible to new linked client contact users.

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/bika/lims/browser/templates/login_details.pt [hunk 1]
```diff
@@ -105,194 +105,207 @@
...
             <div class="formControls" tal:condition="active">
-              <input class="context"
+              <input class="context btn btn-sm btn-warning"
                      type="submit"
                      tabindex=""
                      name="unlink_button"
                      value="Unlink User"
                      i18n:attributes="value"
                      tal:attributes="tabindex tabindex/next;" />
+              <div class="form-text text-muted">
```
Comment (winniehell) [inline src/bika/lims/browser/templates/login_details.pt]: is this missing an i18n attribute?

```suggestion
              <div class="form-text text-muted" i18n:translate="">
```

File: src/bika/lims/browser/templates/login_details.pt [hunk 2]
```diff
@@ -105,194 +105,207 @@
...
+      <!-- /Contact has a linked User -->
+
+      <!-- Contact has **no** linked User -->
+      <tal:withoutuser condition="not:has_user"
+                       define="contactprops view/get_contact_properties;">
+
+        <p class="form-text text-muted" i18n:translate="">
+          No user exists for
+          <span tal:content="here/getFullname" i18n:name="contact_fullname">Pete Smith</span>
+          and he/she will not be able to log in. Fill in the form below to create one for him/her.
```
Comment (winniehell) [inline src/bika/lims/browser/templates/login_details.pt]: random side note: singular they would also be an option here

```suggestion
          and they will not be able to log in. Fill in the form below to create one for them.
```

File: src/bika/lims/browser/templates/login_details.pt [hunk 3]
```diff
@@ -105,194 +105,207 @@
...
-            <label i18n:translate="">Password</label>
-            <span class="fieldRequired" title="Required" i18n:attributes="title" i18n:translate="">(Required)</span>
-            <div class="formHelp" i18n:translate="">
-              Minimum 5 characters.
+
+            <!-- Form Controls -->
+            <div class="formControls">
+              <input class="btn btn-sm btn-primary"
+                     type="submit"
+                     tabindex=""
```
Comment (winniehell) [inline src/bika/lims/browser/templates/login_details.pt:208]: was leaving tabindex empty intentional?

Comment (ramonski) [reply src/bika/lims/browser/templates/login_details.pt:208]: Yes, it could be actually be removed.

**Comment by ramonski** [comment]
Thanks @winniehell for your review and suggestions!

File: src/bika/lims/content/contact.py [hunk 4]
```diff
@@ -303,15 +303,15 @@ def _delUserFromGroup(self, username, group="Clients"):
...
         """
         if hasattr(aq_base(obj), "objectValues"):
             for child_obj in obj.objectValues():
-                self._recursive_reindex_object_security(child_obj)
+                self._recursive_reindex_object(child_obj)
 
-        logger.debug("Reindexing object security for {}".format(repr(obj)))
-        obj.reindexObjectSecurity()
+        logger.debug("Reindexing object {}".format(repr(obj)))
+        obj.reindexObject()
```
Comment (xispa) [inline src/bika/lims/content/contact.py]: `reindexObjectSecurity` forces the reindex of "security" indexes only,
- https://github.com/zopefoundation/Products.CMFCore/blob/361a30e0c72a15a21f88433b8d5fc49331f36728/src/Products/CMFCore/CMFCatalogAware.py#L128 and 
- https://github.com/zopefoundation/Products.CMFCore/blob/361a30e0c72a15a21f88433b8d5fc49331f36728/src/Products/CMFCore/CMFCatalogAware.py#L97

that are those involved on the internal filtering  searches by current user: 
- https://github.com/zopefoundation/Products.CMFCore/blob/361a30e0c72a15a21f88433b8d5fc49331f36728/src/Products/CMFCore/CatalogTool.py#L201-L208

The reindex of `allowedRolesAndUsers` index is way faster than the reindex of all object indexes. Is there any reason I am not aware of that requires the reindex of the whole object for the object to be displayed in listings?

**Review by xispa — CHANGES_REQUESTED** [review]

File: src/bika/lims/content/contact.py (hunk 4 above)
Comment (ramonski) [reply src/bika/lims/content/contact.py]: Ah, I think I mixed it and wondered why it did not work properly!
I will check this and change it to reindex only the `allowedRolesAndUsers` if it works out.

**Review by xispa — APPROVED** [review]

**Pull request merged**